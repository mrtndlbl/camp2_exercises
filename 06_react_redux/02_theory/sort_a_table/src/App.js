import React, { Component } from "react";
import _ from "underscore";
import Row from "./Row";

class Table extends Component {
  constructor(props) {
    super(props);
    this.state = {
      lines: this.props.lines,
      sortedBy: "decathlon_id",
      reverseSort: false
    };

    this.filter = this.filter.bind(this);
  }

  filter(filterBy) {
    if (this.state.sortedBy === filterBy) {
      this.setState({reverseSort: !this.state.reverseSort});
    } else {
      this.setState({
        sortedBy: filterBy,
        reverseSort: false
      });
    }
  }

  render() {
    let sortedLines = _.sortBy(this.state.lines, this.state.sortedBy);
    if (this.state.reverseSort) {
      sortedLines.reverse();
    }
    return (
      <table>
        <TableHead filter={this.filter} />
        {sortedLines.map(Row)}
      </table>
    );
  }
}

export default Table;
