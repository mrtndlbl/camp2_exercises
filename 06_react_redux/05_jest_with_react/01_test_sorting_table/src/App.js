import React, { Component } from "react";
import _ from "underscore";

function Row(props) {
  return (
    <tr key={props.decathlon_id}>
      <td>{props.decathlon_id}</td>
      <td className="title">{props.title}</td>
      <td className="price">{props.price}</td>
    </tr>
  );
}

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      lines: this.props.lines,
      sortedBy: "decathlon_id",
      reverseSort: false
    };
  }

  filter(filterBy) {
    if (this.state.sortedBy === filterBy) {
      this.setState({reverseSort: !this.state.reverseSort});
    } else {
      this.setState({
        sortedBy: filterBy,
        reverseSort: false
      });
    }
  }

  render() {
    let sortedLines = _.sortBy(this.state.lines, this.state.sortedBy);
    if (this.state.reverseSort) {
      sortedLines.reverse();
    }
    return (
      <table>
        <thead>
          <tr>
            <th onClick={() => this.filter("decathlon_id")}>ID</th>
            <th className="titleHeader" onClick={() => this.filter("title")}>Title</th>
            <th className="priceHeader" onClick={() => this.filter("price")}>Price</th>
          </tr>
        </thead>
        <tbody>
          {sortedLines.map(Row)}
        </tbody>
      </table>
    );
  }
}

export default App;
