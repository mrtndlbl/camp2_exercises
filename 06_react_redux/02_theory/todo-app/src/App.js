import React, { Component } from "react";
import "./App.css";

class Todo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: "",
      toDoList: []
    };
  }

  handleInput = event => {
    this.setState({ value: event.target.value, check: false });
  };

  handleSubmit = event => {
    event.preventDefault();
    let newList = this.state.toDoList;
    newList.push({ value: this.state.value, check: false });
    this.setState({ toDoList: newList, value: "" });
  };

  strikeThrough = index => {
    let newToDoList = this.state.toDoList;
    if (newToDoList[index].check === false) {
      newToDoList[index] = {
        value: this.state.toDoList[index].value,
        check: true
      };
    } else {
      newToDoList[index] = {
        value: this.state.toDoList[index].value,
        check: false
      };
    }
    this.setState({ toDoList: newToDoList });
  };

  deleteTask = index => {
    let newToDoList = this.state.toDoList;
    newToDoList.splice(index, 1);
    this.setState({ toDoList: newToDoList });
  };

  render() {
    return (
      <div className="App">
        <h1>Welcome to the ultimate todo app</h1>
        <form onSubmit={this.handleSubmit}>
          <label>Please enter your todo-list below</label>
          <div>
            <input
              type="text"
              id="task"
              value={this.state.value}
              onChange={this.handleInput}
            />
            <input type="submit" value="Submit" />
          </div>
          {this.state.toDoList.map((todo, index) => (
            <div className={"classCheck" + todo.check}>
              <input
                type="checkbox"
                checked={todo.check}
                onChange={() => this.strikeThrough(index)}
              />
              <span>{todo.value}</span>
              {todo.check ? (
                <button type="button" onClick={() => this.deleteTask(index)}>
                  Delete task
                </button>
              ) : null}
            </div>
          ))}
        </form>
      </div>
    );
  }
}

export default Todo;
