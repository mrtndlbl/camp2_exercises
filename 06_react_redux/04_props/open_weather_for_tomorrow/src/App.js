import React, { Component } from 'react';
import './App.css';
import { getWeatherForTomorrow } from "./getWeatherForTomorrow.js";

class WeatherApp extends Component {
  constructor(props) {
    super(props);
    this.state = {
      temp: ""
    }
  }

  componentDidMount() {
    getWeatherForTomorrow()
      .then(newTemp => {
        this.setState({
          temp: newTemp
        });
      });
  }

  render() {
    return (
      <div className="App">
        <h3>Bonjour Monsieur</h3>
        <p>Temperature is {this.state.temp}Â°C</p>
        <h1>branlette()</h1>
        <img src="https://media.giphy.com/media/3ohzdIfd39bRZhKRLa/giphy.gif" width="480" height="480" className="giphy-embed"/>
        <p>
          <a href="https://giphy.com/gifs/funkofficial-hand-3ohzdIfd39bRZhKRLa">via GIPHY</a>
        </p>
      </div>
    );
  }
}

export default WeatherApp;
