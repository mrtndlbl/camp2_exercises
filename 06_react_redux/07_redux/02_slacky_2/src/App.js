import React, { Component } from "react";
import "./App.css";
import { ConnectedLogin } from "./Login";
import { ConnectedChat } from "./Chat";
import { connect } from "react-redux";
import { init } from "./webSockets";

class App extends Component {
  componentDidMount(){
    init();
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Slacky</h1>
        </header>

        {this.props.userName ? <ConnectedChat /> : <ConnectedLogin />}
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    userName: state.loginReducer.userName,
    messages: state.messageReducer.messages
  };
}

export const ConnectedApp = connect(mapStateToProps)(App);
export default ConnectedApp;
