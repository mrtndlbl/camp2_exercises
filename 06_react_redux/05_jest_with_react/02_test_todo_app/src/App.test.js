import React from 'react';
import renderer from 'react-test-renderer';
import App from "./App";
import { configure, shallow, mount  } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

beforeAll(() => {
  configure({ adapter: new Adapter() });
})

test('The toDoList appears at server starting', () => {
  const component = renderer.create(<App />);
  let table = component.toJSON();
  expect(table).toMatchSnapshot();
});

test('We can save a task', () => {
  const component = mount(<App />);
  const newTodo = "First todo";
  component.setState({newTask: newTodo});
  component.find("form").simulate("submit", { preventDefault(){}})
  expect(component.state().tasks).toEqual([{"completed": false, "id": 1, "label": "First todo"}]);
  expect(component.state().tasks.length).toEqual(1);
});

test('We can delete a task', () => {
  const component = mount(<App />);
  const newTodo = "First todo";
  component.setState({newTask: newTodo});
  component.find("form").simulate("submit", {preventDefault(){}});
  component.find(".todos input").simulate("change", {preventDefault(){}});
  component.find(".deleteButton").simulate("click", {preventDefault(){}});
  expect(component.state().tasks.length).toEqual(0);
});
