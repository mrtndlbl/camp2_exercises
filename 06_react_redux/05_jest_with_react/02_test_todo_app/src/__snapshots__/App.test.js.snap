// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The toDoList appears at server starting 1`] = `
<div
  className="Todo"
>
  <div
    className="new-todo"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
  <ul
    className="todos"
  />
</div>
`;
